inspirehep/modules/records/api.py

```diff
@@ -77,24 +80,92 @@ def create(cls, *args, **kwargs):
                 back to the value of the ``RECORDS_SKIP_FILES`` configuration
                 variable.
 
+        Examples:
+            >>> record = {
+            ...     '$schema': 'hep.json',
+            ... }
+            >>> record = InspireRecord.create(record)
+            >>> record.commit()
         """
-        config_skip_files = current_app.config.get('RECORDS_SKIP_FILES')
+        files_src_records = kwargs.pop('files_src_records', [])
+        skip_files = kwargs.pop(
+            'skip_files', current_app.config.get('RECORDS_SKIP_FILES'))
 
-        files_src_records = kwargs.pop('files_src_records', ())
-        skip_files = kwargs.pop('skip_files', config_skip_files)
+        id_ = id_ or uuid.uuid4()
+        data = strip_empty_values(data)
 
-        data = strip_empty_values(*args)
-        new_record = super(InspireRecord, cls).create(data=data, **kwargs)
+        cls.mint(id_, data)
+        record = super(InspireRecord, cls).create(data, id_=id_, **kwargs)
 
         if not skip_files:
-            new_record.download_documents_and_figures(
+            record.download_documents_and_figures(
                 src_records=files_src_records,
             )
-            new_record.commit()
 
-        return new_record
+        return record
+
+    @classmethod
+    def create_or_update(cls, data, **kwargs):
+        """Create or update a record.
+
+        It will check if there is any record registered with the same
+        ``control_number`` and ``pid_type``. If it's ``True``, it will update
+        the current record, otherwise it will create a new one.
+
+        Keyword Args:
+
+            files_src_records(List[InspireRecord]): if passed, it will try to
+                get the files for the documents and figures from the first
+                record in the list that has it in it's files iterator before
+                downloading them, for example to merge existing
+                records.
+            skip_files(bool): if ``True`` it will skip the files retrieval
+                described above. Note also that, if not passed, it will fall
+                back to the value of the ``RECORDS_SKIP_FILES`` configuration
+                variable.
+
+        Examples:
+            >>> record = {
+            ...     '$schema': 'hep.json',
+            ... }
+            >>> record = InspireRecord.create_or_update(record)
+            >>> record.commit()
+        """
 
-    def update(self, *args, **kwargs):
+        pid_type = get_pid_type_from_schema(data['$schema'])
+        control_number = data.get('control_number')
+
+        files_src_records = kwargs.pop('files_src_records', [])
+        skip_files = kwargs.pop(
+            'skip_files', current_app.config.get('RECORDS_SKIP_FILES'))
+
+        try:
+            pid = PersistentIdentifier.get(pid_type, control_number)
+            record = super(InspireRecord, cls).get_record(pid.object_uuid)
```

Comment:
we should pull `get_db_record`, `get_db_records` (as `get_record_by_pid`, `get_records_by_pid`) inside the class as it's more generally useful.